pkg/controller/backup_deletion_controller.go

```diff
@@ -295,34 +295,39 @@ func (c *backupDeletionController) processRequest(req *velerov1api.DeleteBackupR
 		errs = append(errs, err.Error())
 	}
 
-	// Download the tarball
-	backupFile, err := downloadToTempFile(backup.Name, backupStore, log)
-	if err != nil {
-		return errors.Wrap(err, "error downloading backup")
-	}
-	defer closeAndRemoveFile(backupFile, c.logger)
-
 	actions, err := pluginManager.GetDeleteItemActions()
 	log.Debugf("%d actions before invoking actions", len(actions))
 	if err != nil {
 		return errors.Wrap(err, "error getting delete item actions")
 	}
 	// don't defer CleanupClients here, since it was already called above.
 
-	ctx := &delete.Context{
-		Backup:          backup,
-		BackupReader:    backupFile,
-		Actions:         actions,
-		Log:             c.logger,
-		DiscoveryHelper: c.helper,
-		Filesystem:      filesystem.NewFileSystem(),
-	}
+	if len(actions) > 0 {
+		// Download the tarball
+		backupFile, err := downloadToTempFile(backup.Name, backupStore, log)
+
+		if err != nil {
+			log.WithError(err).Errorf("Unable to download tarball for backup %s, skipping associated DeleteItemAction plugins", backup.Name)
+		} else {
+			defer closeAndRemoveFile(backupFile, c.logger)
```

Comment:
Should this `defer` be in the else block? I don't think the `backupFile` variable exists here, unless I'm reading something wrong.

Reply:
It's defined on 307. I put the `defer` here thinking that we shouldn't attempt to close and remove it unless it was successfully created, but I should check the implementation of both functions. `downloadToTempFile` might create the file and not write to it, or `closeAndRemoveFile` might be able to handle the case where it doesn't exist and in that case it would be safe to move it to the outer scope where `backupFile` was defined.

Reply:
`downloadToTempFile` returns a `nil` file pointer in the case where there was an error. This means that `closeAndRemoveFile` will panic as it was passed `nil`. I can leave the `defer` in the `else` block or update `closeAndRemoveFile` to handle the case where the file is `nil`. I'm leaning towards the latter option but let me know what you think!

Reply:
Got it, thanks for the detailed reply. I was actually confusing the `if`s and thought the `else` was part of the `len(actions) > 0` condition.

I think updating `closeAndRemoveFile` to handle a `nil` file is a good idea. That way, we can move the `defer` right after the `downloadToTempFile` call, which is more idiomatic Go, where the `defer` is put right after the creation or fetching of the resource.

The other option is a short comment above the `defer` statement where it's at now to explain why, but I think updating `closeAndRemoveFile` is the better choice.